storage/src/backend/connection.rs

```diff
@@ -165,11 +165,39 @@ impl ProxyHealth {
     }
 }
 
+const UNSET: i16 = 0;
+const REPLACE: i16 = 1;
+const RETAIN: i16 = 2;
+
 #[derive(Debug)]
 struct Proxy {
     client: Client,
     health: ProxyHealth,
     fallback: bool,
+    use_http: bool,
```

Comment:
Is necessary to use this variable? How about reusing `replace_scheme`?

Reply:
We can't because `use_http` is a switch controlling whether to replace HTTP scheme while `replace_sheme` is only a cache indicating whether we should try replacing the scheme if the original registry URL is already HTTP scheme.